react/src/lib/components/SubsurfaceViewer/SubsurfaceViewer.tsx

```diff
@@ -284,7 +285,7 @@ SubsurfaceViewer.propTypes = {
      * Parameters for the Distance Scale component
      * Unit for the scale ruler
      */
-    coordinateUnit: PropTypes.string,
+    coordinateUnit: PropTypes.any,
```

Comment:
What is the difference with `PropTypes.objectOf(PropTypes.any)` ?

Reply:
We have to change this back to `string` and then pretend to have gotten a valid value. The valid values are only defined in TS and can, hence, not be converted to `propTypes` (other way around would work). Writing our own list of valid values would not be sustainable. `dash-generate-components` does, however, support parsing TS files now, so it should be able to extract the types and transfer them to Python. However, I am not sure if it prioritizes `propTypes` or `TS types` when both are available in a file. We can keep this for later.

Reply:
```typescript
convert().possibilities()
```
It is actually possible to get a list of all valid units. We can use `PropTypes.oneOf(...)` to allow only these units at runtime. However, `dash-generate-components` cannot execute functions when parsing component code. The type is automatically casted to `unknown`. I think we can live with this. `PropTypes` should make sure that we always get valid units. In addition, I added a check to `SubsurfaceViewer` - if the unit is invalid, it will log an error and set the unit to undefined.

Reply:
I removed the custom warning as `PropTypes` is already logging a warning to the console.